nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -151,15 +151,28 @@ WaypointFollower::on_shutdown(const rclcpp_lifecycle::State & /*state*/)
   return nav2_util::CallbackReturn::SUCCESS;
 }
 
-template<typename T, typename U, typename V, typename Z>
+template<typename T, typename V, typename Z>
 void WaypointFollower::followWaypointsLogic(
   const T & action_server,
-  const U & poses,
   const V & feedback,
   const Z & result)
 {
   auto goal = action_server->get_current_goal();
 
+  std::vector<geometry_msgs::msg::PoseStamped> poses;
+
+  // compile time static check to decide which block of code to be built
+  if constexpr (std::is_same<T, std::unique_ptr<ActionServer>>::value)
```

Comment:
This looks repeated below. Maybe good to break out into a function so it  can be reused twice?

Reply:
resdolved in https://github.com/ros-planning/navigation2/pull/2111/commits/9c0e4085d9df3e8ca7bb572efa8aa8602482d0da